sdks/python/apache_beam/runners/portability/expansion_service_test.py

```diff
@@ -226,6 +230,27 @@ def from_runner_api_parameter(
     return PartitionTransform()
 
 
+class ExtractHtmlTitleDoFn(beam.DoFn):
+  def process(self, element):
+    from bs4 import BeautifulSoup
+    soup = BeautifulSoup(element, 'html.parser')
+    return [soup.title.string]
+
+
+@ptransform.PTransform.register_urn(TEST_PYTHON_BS4_URN, None)
+class ExtractHtmlTitleTransform(ptransform.PTransform):
+  def expand(self, pcoll):
+    return pcoll | beam.ParDo(ExtractHtmlTitleDoFn()).with_output_types(unicode)
+
+  def to_runner_api_parameter(self, unused_context):
```

Comment:
Bit unrelated but why did we decide to require to_runner_api_parameter/from_runner_api_parameter
instead of builders for using Python from Java ? Or is this just one way of defining Python x-lang transforms ?

Reply:
It's for defining Python ptransform and not specific to x-lang transforms. Python SDK doesn't have builder/registrar for x-lang transforms. In Python SDK, x-lang transforms are just ordinary ptransforms annotated with x-lang compatible urns.